### ISSUE TITLE
parseFormquery can't resolve _path

### ISSUE THREAD

**Issue by fulv** [plone/plone.app.querystring#56]
I'm trying to create a collection with a single criteria - a date range.  When I save, the view throws the traceback below.  I've stepped through `plone.app.querystring.queryparser, line 36, in parseFormquery`, and the `query` contains two criteria:  the date range I requested, plus a `path` from `/`.  Apparently, it doesn't know what to do with the second.

```
2015-10-17 18:09:03 ERROR Zope.SiteErrorLog 1445094543.580.0541094672057 http://localhost:8080/ploneorg/test-folder/test-collection/listing_view
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.Five.browser.metaconfigure, line 485, in __call__
  Module Products.Five.browser.pagetemplatefile, line 125, in __call__
  Module Products.Five.browser.pagetemplatefile, line 59, in __call__
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
  Module zope.pagetemplate.pagetemplate, line 240, in __call__
  Module zope.tal.talinterpreter, line 271, in __call__
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 858, in do_defineMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 946, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 858, in do_defineMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 858, in do_defineMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 583, in do_setLocal_tal
  Module zope.tales.tales, line 696, in evaluate
   - URL: /Users/fulvio/.buildout/eggs/plone.app.contenttypes-1.1b6-py2.7.egg/plone/app/contenttypes/browser/templates/listing.pt
   - Line 22, Column 4
   - Expression: <PathExpr standard:u'view/batch'>
   - Names:
      {'args': (),
       'container': <Collection at /ploneorg/test-folder/test-collection>,
       'context': <Collection at /ploneorg/test-folder/test-collection>,
       'default': <object object at 0x10026dbc0>,
       'here': <Collection at /ploneorg/test-folder/test-collection>,
       'loop': {},
       'nothing': None,
       'options': {},
       'repeat': <Products.PageTemplates.Expressions.SafeMapping object at 0x10f6b1208>,
       'request': <HTTPRequest, URL=http://localhost:8080/ploneorg/test-folder/test-collection/listing_view>,
       'root': <Application at >,
       'template': <Products.Five.browser.pagetemplatefile.ViewPageTemplateFile object at 0x10b361410>,
       'traverse_subpath': [],
       'user': <PropertiedUser 'admin'>,
       'view': <Products.Five.metaclass.SimpleViewClass from /Users/fulvio/.buildout/eggs/plone.app.contenttypes-1.1b6-py2.7.egg/plone/app/contenttypes/browser/templates/listing.pt object at 0x10f5c2490>,
       'views': <Products.Five.browser.pagetemplatefile.ViewMapper object at 0x10f67b790>}
  Module zope.tales.expressions, line 217, in __call__
  Module Products.PageTemplates.Expressions, line 155, in _eval
  Module Products.PageTemplates.Expressions, line 117, in render
  Module plone.app.contenttypes.browser.collection, line 44, in batch
  Module plone.app.contenttypes.browser.collection, line 39, in results
  Module plone.app.contenttypes.behaviors.collection, line 144, in results
  Module plone.app.querystring.querybuilder, line 88, in __call__
  Module plone.app.querystring.querybuilder, line 110, in _makequery
  Module plone.app.querystring.queryparser, line 36, in parseFormquery
  Module zope.dottedname.resolve, line 38, in resolve
ImportError: No module named _path
```

Here are some of the version pins:
```
plone.app.contenttypes = 1.1b6
plone.app.relationfield = 1.2.2
plone.app.z3cform = 1.0.1
plone.dexterity = 2.2.3
plone.app.portlets = 2.5.2
plone.formwidget.querystring = 1.1.5
Products.DateRecurringIndex = 2.1
collective.elephantvocabulary = 0.2.5
icalendar = 3.9.0
plone.app.event = 1.1.5
plone.event = 1.1
plone.formwidget.recurrence = 1.2.6
plone.outputfilters = 1.13
z3c.form = 3.2.1
plone.app.jquery = 1.8.3
plone.app.widgets = 1.8.0
plone.app.querystring = 1.3.8
```

Site is Plone 4.3.7

**Comment by rnixx** [comment]
https://github.com/plone/plone.app.querystring/blob/master/plone/app/querystring/queryparser.py#L27

seems that "%s.operation" maps to "queryparser._path" function, which not exists any longer.

Does this error happen in an existing plone site after upgrading? If so, can you retry with a vanilla plone instance?

**Comment by fulv** [comment]
On a vanilla plone site it does not happen, only after upgrading.

**Comment by mauritsvanrees** [comment]
By default Plone 4.3.7 is using plone.app.querystring 1.2.7.  You are using 1.3.8.  Any reason for that?

In 1.2.7 `queryparser.py` still has `_path` in it, so that version should work.

If you have to stick with 1.3.8: maybe there is an upgrade step when you go to the portal_setup Upgrades tab.

**Comment by pbauer** [comment]
@fulv if you import the complete profile of plone.app.querystring (after running all upgrade-steps!) that issue is resolved.
@mauritsvanrees I currently try to use p.a.querystring 1.3.11 in Plone 4.3.6. Mainly because I need the "any"-operator for keywords. So far I did not find any issues (except for the above). Do you know the reason why we use the branch 1.2.x in Plone 4?

**Comment by mauritsvanrees** [comment]
Well, the path related queries had changes in 1.3(.0) that may have been deemed too aggressive for including in Plone 4.3.x.
See https://github.com/plone/plone.app.querystring/pull/29
The version was updated to 1.3.dev0 in that pull request.

**Comment by jensens** [comment]
I think this is fixed meanwhile with #29 - please reopen if I'am wrong.
